### PR TITLE
Clarify naming and descriptions, differentiate prompt requests and access decisions, and fix some response formatting

### PR DESCRIPTION
**Pull request by olivercalder**
This PR includes various improvements as discussed with @mvo5, @tim-hm, and @pedronis, including:

- Clarified naming of "prompt requests" and "access decisions":
  - "prompt request" differentiates it from HTTP requests
  - "access decisions" separates stored decisions somewhat from the prompt requests which may or may not have led to their creation -- access decisions can be created directly via `POST /v2/prompting/decisions`
- Replaced "response" verbiage with "reply" so that prompt request replies are distinct from the HTTP meaning of "response"
- Moved both prompt request replies modified decision contents out of the query parameter and into their respective response bodies
  - This was incorrect before -- thanks @pedronis
- Renamed `path-access` to `path-scope`
- Renamed `duration` to `lifetime`
- Removed the `stored-decisions` schema and the `snap-param`, `app-param`, and `follow-param` parameters and replaced each with more specific hard-coded parameters with better descriptions
- Added comments about when the `app` and `follow` parameters are ignored in the absence of the `snap` parameter for the `/v2/prompting/decisions` endpoint
  - Now, the `follow` parameter can only be used on `/v2/prompting/decisions` if the `snap` parameter is also included, to avoid pagination problems
- Made it harder to accidentally delete many access decisions:
  - Restricted the `DELETE /v2/prompting/decisions` method to require the `snap` parameter, thus only allowing deletion of at most all decisions for a single snap at a time
  - Added the required `confirm-delete` parameter to the `DELETE /v2/prompting/decisions` method which the client must deliberately include in order to delete multiple decisions

Some changes still to consider:

- Should access decisions be modifiable by re-sending a `reply` schema or sending a `decision-contents` schema?
  - That is, should fundamental properties of an access decision (the `snap`, `app`, `path`, and `resource-type` fields) be modifiable?
  - The `reply` schema only has the `allow`, `lifetime`, `permissions`, and `path-scope` fields (also included in `decision-contents`)
  - If we only want to allow modifying `allow`, `lifetime`, `permissions`, and `path-scope`, should we make a generalized schema not called `reply` which can be used both by prompt request replies and access decision modifications?
- The `resource-type` schema is not named well... what should it be called instead?
- Should we use the name `operation` in place of `permission` in the API and internally within snapd when referring to read, write, execute, etc.?

### CODE REVIEW

**Comment by olivercalder** [comment]
Another thing to consider: should the current lists of decisions returned by various endpoints be replaced by json objects indexed by ID? That is, maps from `id` to `decision-contents`? This would reinforce the hierarchy of decision IDs with "contained" decision information.

This can be done by defining `additionalProperties` which map to a particular type, but `patternProperties` can be used instead to only allow keys matching a given regex.  If we define the format of decision IDs, we could safely restrict the format of keys in the json structure.

Regardless of whether `patternProperties` or `additionalProperties` is used, should the returned list of decisions be "un-flattened" and returned by-ID?

File: README.md [hunk 1]
```diff
@@ -145,77 +191,85 @@ Modify the stored decision with the given ID.
...
 
 - `id`: The unique identifier of the stored decision
-- `response`: The updated [`response`](#response) information to store with the decision
 
-##### Returns
+##### Request Body
+
+The updated [`reply`](#reply) information to replace that which was previously associated with the decision.
+
+Since a `reply` object is used as the payload, the `allow` (allow/deny), `lifetime`, `permissions` (list of operations like `read`, `write`, etc.), and `path-scope` (file, directory, subdirectories) fields of the stored access decision can be modified.
```
Comment (tim-hm) [inline README.md]: I wonder if `lifespan` is better than `lifetime`

Comment (olivercalder) [reply README.md]: Yes it is. That's the Rust brain getting to me. Fixed in new commit, thanks!

**Comment by olivercalder** [comment]
More structural changes were proposed at the meeting last week to better correspond with the existing snapd API, so it is likely that this repo will be deprecated in favor of specifications defined in https://docs.google.com/document/d/1tBnefdukP69EUJOlH8bgD2hrvZCYoE8-1ZlqRRYlOqc

For the sake of completeness, I'll merge this PR into main now, and possibly open a new PR with the more structural changes if it seems that doing so would be useful.

**Pull request merged**